## Zadatak 1

In [7]:
# m' = 311

for i in range(100):
    T = 2001 + i
    U = T*311 % 1000
    
    if (T + U*209)/1000 >= 209:
        print(T)
        break

2045


Traženi T je jednak **2045**.

## Zadatak 2

In [14]:
# Garnerov algoritam za CRT
k = 3
mi = [1,1]
M = [7,11,13] 
c = [0,0]
X = [2,2,12]
MM = 7*11*13

for i in range( k - 1 ):
    for j in range(i+1):
        mi[i] *= M[j]
    c[i] = pow(mi[i], -1, M[i+1])
    
x = X[0]
for i in range(k-1):
    y = ((X[i+1] - x)*c[i]) % M[i+1]
    x = x + y * mi[i]
    
x = x % MM
print(x)

233


Rješenje je $233 + k \cdot 7 \cdot 11 \cdot 13$, $k \in \mathbb{N} \cup \{0\}$. 

## Zadatak 3

In [17]:
import math

In [69]:
def verizni_razlomak1( alpha, s_i, t_i, d):
    
    rjesenje = []
    S = [s_i]
    T = [t_i]
    i = 1
    
    while(1):
        a_i = int(alpha)
        rjesenje.append(a_i)
        s_ii = a_i * t_i - s_i
        t_ii = (d - s_ii**2) / t_i
        S.append(s_ii)
        T.append(t_ii)
    
        alpha = (s_ii + math.sqrt(d)) / t_ii
        t_i = t_ii
        s_i = s_ii
        
        if( i > 1 and S[1] == S[i] and T[1] == T[i] ):
            break
        i += 1
        if(i > 17):
            break
    
    if( i == 17 ):
        print(d)
        print(*rjesenje)

In [70]:
for i in range(100):
    if i != 0 and int(math.sqrt(i)) != math.sqrt(i):
        verizni_razlomak1(math.sqrt(i), 0, 1, i)

94
9 1 2 3 1 1 5 1 8 1 5 1 1 3 2 1 18


Najmanji takav d je jednak **94**.